src/slices/auth.slice.js

```diff
@@ -0,0 +1,78 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit'
+import { API_LOGIN, API_SIGNUP } from 'utils'
+import axios from 'axios'
+
+const initialState = {
+    isUserLoggedIn: false,
+    isUserSignedUp: false,
+    error: '',
+}
+
+export const logInUser = createAsyncThunk(
+    'auth/logInUser',
+    async ({ username, password }) => {
+        const response = await axios.post(API_LOGIN, {
+            username,
+            password,
+        })
+        return response
+    }
+)
+
+export const signUpUser = createAsyncThunk(
+    'auth/signUpUser',
+    async ({ username, password, firstName, lastName }) => {
+        const response = await axios.post(API_SIGNUP, {
+            username,
+            password,
+            firstName,
+            lastName
+        })
+        return response
+    }
+)
+
+export const authSlice = createSlice({
+    name: 'auth',
+    initialState,
+    reducers: {
+        logOutUser: (state) => {
+            window.localStorage.removeItem('userToken')
+            window.localStorage.removeItem('username')
+            state.isUserLoggedIn = false
+        },
+        updateUserLogInStatus: (state) => {
+            state.isUserLoggedIn = true
+        }
+    },
+    extraReducers: {
+        [logInUser.fulfilled]: (state, action) => {
+            window.localStorage.setItem('userToken', action.payload.data.encodedToken)
+            window.localStorage.setItem('username', action.payload.data.foundUser.username)
+            state.isUserLoggedIn = true
```

Comment:
I think all local storage managing should be moved to thunk as it as a side effect

Reply:
I've made the changes. Thanks.